csr/csr.go

```diff
@@ -31,12 +31,12 @@ var badSignatureAlgorithms = map[x509.SignatureAlgorithm]bool{
 }
 
 var (
-	invalidPubKey       = errors.New("invalid public key in CSR")
-	unsupportedSigAlg   = errors.New("signature algorithm not supported")
-	invalidSig          = errors.New("invalid signature on CSR")
-	invalidEmailPresent = errors.New("CSR contains one or more email address fields")
-	invalidIPPresent    = errors.New("CSR contains one or more IP address fields")
-	invalidNoDNS        = errors.New("at least one DNS name is required")
+	invalidPubKey       = berrors.MalformedError("invalid public key in CSR")
```

Comment:
Ah, good point. I had missed that. However, that makes this a substantive change in what sort of errors we return in certain cases, which is easy to lose in a big change like this one. Could you pull out these changes, and the corresponding ra.go change to not always wrap in Malformed, into a follow-on PR, with tests for the newly returned error types?